### PR TITLE
Make ES security codeowners for privilege definitions

### PR DESCRIPTION
**Pull request by n1v0lg**
This PR makes the ES security code-owners of privilege-related source files. This covers named cluster and index privileges. We may extend code-ownership to other security-related files, but this feels like a reasonable start.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -59,3 +59,6 @@ server/src/main/java/org/elasticsearch/bootstrap @elastic/es-core-infra
 server/src/main/java/org/elasticsearch/node @elastic/es-core-infra
 server/src/main/java/org/elasticsearch/plugins @elastic/es-core-infra
 server/src/main/java/org/elasticsearch/threadpool @elastic/es-core-infra
+
+# Security
+x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege @elastic/es-security
```
Comment (n1v0lg) [inline .github/CODEOWNERS:64]: This will only apply once it's merged, so can't "test" and add this in the same PR unfortunately.

**Review by slobodanadamovic — APPROVED** [review]
LGTM 👍

**Review by AthenaEryma — APPROVED** [review]
Great minds think alike @n1v0lg - I opened https://github.com/elastic/elasticsearch/pull/108591 a couple hours after you opened this. I went a bit narrower in scope, but I'm fully on board with putting the whole directory in. I do think `x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java` is probably good to add as well - possibly the whole directory again, but at least that file.

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Pull request merged**